### PR TITLE
sys/net/sock: introduce SOCK_HAS_IPV4

### PR DESCRIPTION
**Pull request by benpicco**
### Contribution description

Add a define analogous to `SOCK_HAS_IPV6` so we can skip compilation of IPv4 related code if only IPv6 is used.

### CODE REVIEW

File: sys/net/sock/sock_util.c [hunk 1]
```diff
@@ -43,11 +43,13 @@ int sock_tl_ep_fmt(const struct _sock_tl_ep *endpoint,
...
+#if defined(SOCK_HAS_IPV4)
         case AF_INET:
             {
                 addr_ptr = (void*)&endpoint->addr.ipv4;
                 break;
             }
+#endif
 #if defined(SOCK_HAS_IPV6)
         case AF_INET6:
             {
```
Comment (maribu) [inline sys/net/sock/sock_util.c:55]: Wouldn't a simple

```C
    if (IS_ACTIVE(SOCK_HAS_IPV4) && (endpoint->family == AF_INET)) {
        addr_ptr = &endpoint->addr.ipv4;
    }
    else if (IS_ACTIVE(SOCK_HAS_IPV6) && (endpoint->family == AF_INET6)) {
        addr_ptr = &endpoint->addr.ipv6;
    }
    else {
        return -ENOTSUP;
    }
```

or maybe even a

```C
    addr_ptr = &endpoint->addr;
    if (!(((endpoint->family == AF_INET) && (IS_ACTIVE(SOCK_HAS_IPV4))
            || ((endpoint->family == AF_INET6) && (IS_ACTIVE(SOCK_HAS_IPV46)))) {
        return -ENOTSUP;
    }
```

be easier?

**Review by maribu — COMMENTED** [review]
Looks good to me. One comment inline. Care to sneak in a style fix commit to make the linters happy?

File: sys/net/sock/sock_util.c (hunk 1 above)
Comment (benpicco) [reply sys/net/sock/sock_util.c:55]: Not sure if it's easier on the eyes too :smile:

**Review by maribu — APPROVED** [review]
lgtm, thx!

**Pull request merged**